R/get_feature_wfs.R

```diff
@@ -81,57 +90,100 @@ get_feature_wfs <- function(
-                      version = version,
-                      typeNames = layername,
-                      srsName = crs,
-                      bbox = bbox,
-                      filter = filter,
-                      cql_filter = cql_filter,
-                      outputFormat = output_format,
-                      propertyName = property_name,
-                      resultType = result_type,
-                      ...
+    url$query <- list(
+      service = "wfs",
+      request = "GetFeature",
+      version = version,
+      typeNames = layername,
+      srsName = crs,
+      bbox = bbox,
+      filter = filter,
+      cql_filter = cql_filter,
+      outputFormat = output_format,
+      propertyName = property_name,
+      resultType = result_type,
+      ...
     )
   }
   if (grepl(pattern = "^1", x = version)) {
-    url$query <- list(service = "wfs",
-                      request = "GetFeature",
-                      version = version,
-                      typeName = layername,
-                      srsName = crs,
-                      bbox = bbox,
-                      filter = filter,
-                      cql_filter = cql_filter,
-                      outputFormat = output_format,
-                      propertyName = property_name,
-                      resultType = NULL,
-                      ...
+    url$query <- list(
+      service = "wfs",
+      request = "GetFeature",
+      version = version,
+      typeName = layername,
+      srsName = crs,
+      bbox = bbox,
+      filter = filter,
+      cql_filter = cql_filter,
+      outputFormat = output_format,
+      propertyName = property_name,
+      resultType = NULL,
+      ...
     )
   }
 
   request <- build_url(url)
 
-  result <- GET(request)
-  parsed <- as_list(content(result, "parsed", encoding = "UTF-8"))
-
-  if (names(parsed) == "ExceptionReport") {
-    message <- unlist(parsed$ExceptionReport$Exception$ExceptionText)
-    old_op <- options(warning.length = max(nchar(message), 1000))
-    on.exit(options(old_op))
-    stop(sprintf(paste0(message, "\nThe requested url was: %s"),
-                 request))
-  }
+  get_result <- GET(request)
+  handle_result_types(
+    get_result,
+    result_type = result_type,
+    property_name = property_name,
+    request = request
+  )
+}
 
-  if (result_type == "hits") {
-    n_features <- attr(parsed$FeatureCollection, "numberMatched")
-    return(n_features)
+handle_result_types <- function(result, result_type, property_name, request) {
+  if (result$status_code == 200L) {
+    if (result_type == "hits") {
+      parsed <- as_list(content(result, "parsed", encoding = "UTF-8"))
+      n_features <- attr(parsed$FeatureCollection, "numberMatched")
+      return(n_features)
+    } else {
+      content <- content(result, encoding = "UTF-8")
+      # Write the content to disk
+      destfile <- paste0(tempfile(), ".gml")
+      if (inherits(content, "xml_document")) {
+        xml2::write_xml(content, destfile)
```

Comment:
Import `write_xml()`
```suggestion
        write_xml(content, destfile)
```

Reply:
see [efa54d6](https://github.com/inbo/inbospatial/pull/1/commits/efa54d606d348cbb4514f97dbd014021634ce692)